python-sdk/nuscenes/map_expansion/arcline_path_utils.py

```diff
@@ -0,0 +1,288 @@
+    scaled_angle = (angle_in_radians - interval_min) % two_pi + interval_min
+    return scaled_angle
+
+
+def compute_segment_sign(arcline_path: ArcLinePath) -> Tuple[int, int, int]:
+    """
+    Compute the sign of an arcline path based on its shape.
+    :param arcline_path: arcline path record.
+    :return: Tuple of signs for all three parts of the path. 0 if straight, -1 if right,
+        1 if left.
+    """
+    shape = arcline_path['shape']
+    segment_sign = [0, 0, 0]
+
+    if shape in ("LRL", "LSL", "LSR"):
+        segment_sign[0] = 1
+    else:
+        segment_sign[0] = -1
+
+    if shape == "RLR":
+        segment_sign[1] = 1
+    elif shape == "LRL":
+        segment_sign[1] = -1
+    else:
+        segment_sign[1] = 0
+
+    if shape in ("LRL", "LSL", "RSL"):
+        segment_sign[2] = 1
+    else:
+        segment_sign[2] = -1
+
+    return segment_sign[0], segment_sign[1], segment_sign[2]
+
+
+def get_transformation_at_s(pose: Pose,
+                            s: float) -> Pose:
+    """
+    Get the affine transformation at s meters along the path.
+    :param pose: Pose represented as tuple (x, y, yaw).
+    :param s: Length along the arcline path in range (0, length_of_arcline_path].
+    :return: Transformation represented as pose tuple.
+    """
+
+    theta = pose[2] * s
+    ctheta = math.cos(theta)
+    stheta = math.sin(theta)
+
+    if abs(pose[2]) < 1e-6:
+        return pose[0] * s, pose[1] * s, theta
+    else:
+        new_x = (pose[1] * (ctheta - 1.0) + pose[0] * stheta) / pose[2]
+        new_y = (pose[0] * (1.0 - ctheta) + pose[1] * stheta) / pose[2]
+        return new_x, new_y, theta
+
+
+def apply_affine_transformation(pose: Pose,
+                                transformation: Pose) -> Pose:
+    """
+    Apply affine transformation to pose.
+    :param pose: Starting pose.
+    :param transformation: Affine transformation represented as a pose tuple.
+    :return: Pose tuple - the result of applying the transformation to the starting pose.
+    """
+
+    new_x = math.cos(pose[2]) * transformation[0] - math.sin(pose[2]) * transformation[1] + pose[0]
+    new_y = math.sin(pose[2]) * transformation[0] + math.cos(pose[2]) * transformation[1] + pose[1]
+    new_yaw = principal_value(pose[2] + transformation[2])
+
+    return new_x, new_y, new_yaw
+
+
+def _get_lie_algebra(segment_sign: Tuple[int, int, int],
+                     radius: float) -> List[Tuple[float, float, float]]:
+    """
+    Gets the Lie algebra for an arcline path.
+    :param segment_sign: Tuple of signs for each segment in the arcline path.
+    :param radius: Radius of curvature of the arcline path.
+    :return: List of lie algebra poses.
+    """
+
+    return [(1.0, 0.0, segment_sign[0] / radius),
+            (1.0, 0.0, segment_sign[1] / radius),
+            (1.0, 0.0, segment_sign[2] / radius)]
+
+
+def pose_at_length(arcline_path: ArcLinePath,
+                   l: float) -> Tuple[float, float, float]:
```

Comment:
`l` is not a recommended variable name. Why not call it `pos` ?

Reply:
Done.